Makefile

```
@@ -154,11 +136,17 @@ bin/%/submariner-operator: $(VENDOR_MODULES) main.go $(EMBEDDED_YAMLS)
 
 ci: $(EMBEDDED_YAMLS) golangci-lint markdownlint unit build images
 
-# Operator CRDs
+# Download controller-gen locally if not already downloaded.
+CONTROLLER_TOOLS_VERSION := 0.9.2
 $(CONTROLLER_GEN): $(VENDOR_MODULES)
 	mkdir -p $(@D)
-	$(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
+	test -s $(CURDIR)/bin/controller-gen || $(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
```

If `controller-gen` should be built only if it doesn’t exist, the prerequisites should be updated; testing for the existence of the binary is what Make does.

```suggestion
$(CONTROLLER_GEN):
	mkdir -p $(@D)
	$(GO) build -o $@ sigs.k8s.io/controller-tools/cmd/controller-gen
```